enterprise/api_client/lms.py

```diff
@@ -398,6 +398,65 @@ def _get_results(self, identity_provider, param_name, param_value, result_field_
         return None
 
 
+class ThirdPartyAuthApiClientJwt(JwtLmsApiClient):
+    """
+    Object builds an API client to make calls to the Third Party Auth API.
+    """
+
+    API_BASE_URL = settings.LMS_INTERNAL_ROOT_URL + '/api/third_party_auth/v0/'
+
+    @JwtLmsApiClient.refresh_token
+    def get_remote_id(self, identity_provider, username):
+        """
+        Retrieve the remote identifier for the given username.
+
+        Args:
+        * ``identity_provider`` (str): identifier slug for the third-party authentication service used during SSO.
+        * ``username`` (str): The username ID identifying the user for which to retrieve the remote name.
+
+        Returns:
+            string or None: the remote name of the given user.  None if not found.
+        """
+        return self._get_results(identity_provider, 'username', username, 'remote_id')
+
+    @JwtLmsApiClient.refresh_token
+    def get_username_from_remote_id(self, identity_provider, remote_id):
+        """
+        Retrieve the remote identifier for the given username.
+
+        Args:
+        * ``identity_provider`` (str): identifier slug for the third-party authentication service used during SSO.
+        * ``remote_id`` (str): The remote id identifying the user for which to retrieve the usernamename.
+
+        Returns:
+            string or None: the username of the given user.  None if not found.
+        """
+        return self._get_results(identity_provider, 'remote_id', remote_id, 'username')
+
+    def _get_results(self, identity_provider, param_name, param_value, result_field_name):
+        """
+        Calls the third party auth api endpoint to get the mapping between usernames and remote ids.
+        """
+        try:
+            kwargs = {param_name: param_value}
+            returned = self.client.providers(identity_provider).users.get(**kwargs)
```

Comment:
`response = self.client.providers(identity_provider).users.get(**query_parms)`

Reply:
This seems to be a reasonable change but the `_get_results` method is being called at two different places, one passes 'remote_id' and its value as `param_name` and `param_value` and the other passes `username` and its value in the same arguments. We'll eventually have to declare these two variables(param_name+param_value) anyway inside `_get_results` to be used here:

https://github.com/edx/edx-enterprise/blob/master/enterprise/api_client/lms.py#L389
https://github.com/edx/edx-enterprise/blob/master/enterprise/api_client/lms.py#L396

Reply:
yes, then you can go with your solution.